### ISSUE TITLE
Call to a member function delete_notes() on null

### ISSUE THREAD

**Issue by PrinceOfAbyss** [Automattic/woocommerce-payments#7110]
### Describe the bug
I tried to deactivate the plugin due to another issue, but during de-activation, the e-shop goes down with `PHP Fatal error:  Uncaught Error: Call to a member function delete_notes() on null in /wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php:1386`

I had to manually rename the plugin's folder to bring back the e-shop online.

### To Reproduce
I suppose you just have to try de-activating the eshop?

### Actual behavior
N/A

### Screenshots
N/A

### Expected behavior
It should de-activate the plugin normally I suppose...

### Desktop (please complete the following information):
Not relevant

### Smartphone (please complete the following information):
Not relevant

### Additional context
N/A

**Comment by zmaglica** [comment]
Greetings,

Thank you for reporting the issue. I've reviewed the function name in question and its call locations. Based on the current version of the plugin, the error you encountered seems unusual, as the file where this service is instantiated is already included. Could you kindly specify which version of the WooCommerce Payments plugin you are using?

For additional context, I also conducted tests by deactivating the plugin on my end and was unable to reproduce the error you mentioned.

**Comment by PrinceOfAbyss** [comment]
Thanks a lot for your prompt reply. [Here](https://www.youtube.com/watch?v=zBThzMHESyQ) is a short video showing the error in action. About the e-shops versions:

WooCommerce version: 		7.7.0
WordPress version: 		6.2.2
PHP version: 		8.0.28

The error log that gets produced when the error occurs reads the following:

```
[04-Sep-2023 09:30:20 UTC] PHP Fatal error:  Uncaught Error: Call to a member function delete_notes() on null in /public_html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php:1386
Stack trace:
#0 /public_html/wp-content/plugins/woocommerce-payments/woocommerce-payments.php(57): WC_Payments::remove_woo_admin_notes()
#1 /public_html/wp-includes/class-wp-hook.php(308): wcpay_deactivated()
#2 /public_html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#3 /public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#4 /public_html/wp-admin/includes/plugin.php(814): do_action()
#5 /public_html/wp-admin/plugins.php(209): deactivate_plugins()
#6 {main}
  thrown in /public_html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php on line 1386
```

**Comment by zmaglica** [comment]
Thank you for the video. It really helped a lot. I tested with brand-new WooCommerce installation with PHP 8.0 installed, and I was still unable to reproduce it. Perhaps there could be a conflict between different plugins that uses predefined hooks for plugin uninstallation. I will move this issue for prioritization, and from there we will define when to tackle this issue.

**Comment by PrinceOfAbyss** [comment]
Thank you!

**Comment by naman03malhotra** [comment]
Maybe solved by https://github.com/Automattic/woocommerce-payments/pull/7148

**Comment by jessy-p** [comment]
Fixed by 6.4.1 patch release.
